### PR TITLE
Implement create group screen

### PR DESCRIPTION
**Pull request by tung2744**
ref #3910 
requires https://github.com/authgear/authgear-server/pull/3963

Screenshot 2024-03-12 at 8 02 12 PM

### CODE REVIEW

**Comment by Ngkaokis** [comment]
@tung2744 Please rebase :adore2x:

File: portal/src/graphql/adminapi/AddGroupScreen.tsx [hunk 1]
```diff
@@ -1,7 +1,206 @@
...
+
+  useEffect(() => {
+    if (form.submissionResult != null) {
+      const groupID = form.submissionResult;
+      navigate(`../${encodeURIComponent(groupID)}/details`, { replace: true });
+    }
+  }, [form.submissionResult, navigate]);
+
+  return (
+    <RoleAndGroupsLayout breadcrumbs={breadcrumbs}>
```
Comment (Ngkaokis) [inline portal/src/graphql/adminapi/AddGroupScreen.tsx]: headerBreadcrumbs

File: portal/src/graphql/adminapi/AddGroupScreen.tsx [hunk 2]
```diff
@@ -1,7 +1,206 @@
...
+        throw new Error("unexpected validation errors");
+      }
+      return createGroup({
+        key: sanitizedRole.key,
+        name: sanitizedRole.name,
+        description: sanitizedRole.description,
+      });
+    },
+    [createGroup]
+  );
```
Comment (Ngkaokis) [inline portal/src/graphql/adminapi/AddGroupScreen.tsx:181]: We have the requirement of auto fill the key field as name if key is not provided

https://linear.app/authgear/issue/DEV-404/roles-and-groups-configuration-in-portal#comment-e5e223c7

Comment (tung2744) [reply portal/src/graphql/adminapi/AddGroupScreen.tsx:181]: I should have done it in [8f4e839](https://github.com/authgear/authgear-server/pull/3966/commits/8f4e83908a1ba3c9629a9d937276f46ea5ce894b)

Comment (tung2744) [reply portal/src/graphql/adminapi/AddGroupScreen.tsx:181]: But I found that the variable name here is wrong due to copy and paste error. Let me fix it.

File: portal/src/model/group.ts [hunk 3]
```diff
@@ -1,5 +1,67 @@
...
+const MAX_KEY_LENGTH = 40;
+
+export function generateGroupKeyFromName(name: string): string {
+  let processedName = name.trim();
+  // Replace all whitespace, -, + with _
+  processedName = processedName.replace(/[\s\-+]/g, "_");
+  // Remove all unallowed characters
+  processedName = processedName.replace(/[^a-zA-Z0-9:_]/g, "");
+  // Remove all unallowed prefix characters
+  processedName = processedName.replace(/^[^a-zA-Z_]*/, "");
```
Comment (Ngkaokis) [inline portal/src/model/group.ts:18]: Any source for the processing?

Comment (tung2744) [reply portal/src/model/group.ts:18]: No source, I just try to convert the name into a format which backend won't return validation error.

Comment (Ngkaokis) [reply portal/src/model/group.ts:18]: Ok, will merge first.

@fungc-io See if the processing look ok to you
in some extreme example
name `123 testing +++ $$$$` -> key `_testing_____`

Comment (fungc-io) [reply portal/src/model/group.ts:18]: ok

**Pull request merged**